src/foam/core/FObject.js

```diff
@@ -808,6 +808,12 @@ foam.CLASS({
 
     function compareTo(other) {
       if ( other === this ) return 0;
+
+      // Use comparator of other if it is not an FObject.
+      if ( ! foam.core.FObject.isInstance(other) ) {
```

Comment:
I think it can be simplified to `if (...other not FObj...) return -foam.util.compare(other, this);`. 
We cannot write `return foam.util.compare(this, other);` or we get an may run into an infinite loop (this is due to the fact FObject is an Object and it causes complications on this matter).

I have kept this as is for now as I'm not sure if that avoids all potential infinite loops as well